src/tools/bin/plot_chargeConservation.py

```diff
@@ -52,55 +51,60 @@ def set_colorbar(cb):
         t.set_fontsize(16)
 
 
-def plotError(h5file, slice_pos=[0.5, 0.5, 0.5]):
+def plotError(file_pattern, slice_pos=[0.5, 0.5, 0.5], timestep=-1):
     """
-    read field data from hdf5 files
+    read field data from a openPMD file
     compute div(E) - rho/epsilon_0
     plot slices through simulation volume
 
     Parameters:
-    h5file: file name
-        file name to hdf5 data set from PIConGPU
+    file_pattern: file name
+         openPMD file series pattern e.g. simData_%%T.bp
 
     slice_pos: list of floats
         list of 3 floats to define slice position [0, 1]
         Default=[0.5, 0.5, 0.5]
     """
     # load hdf5 file
-    f = h5py.File(h5file, "r")
+    series = io.Series(file_pattern, io.Access.read_only)
 
     # read time step (python 2 and 3 save)
-    timestep = -1
-    for i in f['/data'].keys():
-        timestep = i
+    if timestep == -1:
+        for i in series.iterations:
+            timestep = i
+
+    f = series.iterations[timestep]
 
     # load physics constants and simulation parameters
-    EPS0 = f["/data/{}".format(timestep)].attrs["eps0"]
-    CELL_WIDTH = f["/data/{}".format(timestep)].attrs["cell_width"]
-    CELL_HEIGHT = f["/data/{}".format(timestep)].attrs["cell_height"]
-    CELL_DEPTH = f["/data/{}".format(timestep)].attrs["cell_depth"]
+    EPS0 = f.get_attribute("eps0")
+    CELL_WIDTH = f.get_attribute("cell_width")
+    CELL_HEIGHT = f.get_attribute("cell_height")
+    CELL_DEPTH = f.get_attribute("cell_depth")
 
     # load electric field
-    Ex = np.array(f["/data/{}/fields/E/x".format(timestep)])
-    Ey = np.array(f["/data/{}/fields/E/y".format(timestep)])
-    Ez = np.array(f["/data/{}/fields/E/z".format(timestep)])
+    Ex = f.meshes["E"]["x"][:]
+    Ey = f.meshes["E"]["y"][:]
+    Ez = f.meshes["E"]["z"][:]
+
+    series.flush()
 
     # load and add charge density
     charge = np.zeros_like(Ex)
     norm = 0.0
-    for field_name in f["/data/{}/fields/".format(timestep)].keys():
-        if field_name[-14:] == "_chargeDensity":
+
+    for fieldName in f.meshes:
+        if fieldName[-14:] == "_chargeDensity":
```

Comment:
I guess this 14 is length of `_chargeDensity`, maybe better to express like that? So that if the name changes that 14 is not left out